pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -299,10 +327,18 @@ func (c *Controller) injectPacket(tf *opsv1alpha1.Traceflow) error {
 			dstIP = dstPod.Status.PodIP
 			dstNodeIP = dstPod.Status.HostIP
 		}
+	} else if tf.Spec.Destination.Service != "" {
+		dstSvc, err := c.serviceLister.Services(tf.Spec.Destination.Namespace).Get(tf.Spec.Destination.Service)
+		if err != nil {
+			return err
+		}
+		dstIP = dstSvc.Spec.ClusterIP
 	}
-	if dstNodeIP != "" {
+	// Check encap status if Node of destination Pod is remote or destination is IP/Service
+	if dstNodeIP != "" || tf.Spec.Destination.Pod == "" {
 		peerIP := net.ParseIP(dstNodeIP)
-		if c.networkConfig.TunnelType == ovsconfig.GeneveTunnel && peerIP != nil && c.networkConfig.TrafficEncapMode.NeedsEncapToPeer(peerIP, c.nodeConfig.NodeIPAddr) {
+		if c.networkConfig.TunnelType == ovsconfig.GeneveTunnel && (tf.Spec.Destination.Pod == "" || c.networkConfig.TrafficEncapMode.NeedsEncapToPeer(peerIP, c.nodeConfig.NodeIPAddr)) {
+			// Destination is IP/Service or packet will be encapsulated to remote Node
```

Comment:
could you make the comment a complete sentence? hard to understand until I read several times. I think you mean:
If the destination is IP/Service or the packet will be encap...., wait a small ...

Reply:
Change to:
// If the destination is Service/IP or the packet will be encapsulated to remote Node, wait a small period for other Nodes.